kernel/src/expressions/scalars.rs

```diff
@@ -47,7 +47,7 @@ impl Scalar {
             Self::TimestampNtz(_) => DataType::TIMESTAMP_NTZ,
             Self::Date(_) => DataType::DATE,
             Self::Binary(_) => DataType::BINARY,
-            Self::Decimal(_, precision, scale) => DataType::decimal(*precision, *scale),
+            Self::Decimal(_, precision, scale) => DataType::decimal_unchecked(*precision, *scale),
```

Comment:
we use a "special" version `decimal_unchecked` here, to keep the code path infallible - will panic if invalid data is encountered.